### ISSUE TITLE
React Native support

### ISSUE THREAD

**Issue by gsgualbano** [christianalfoni/stitches#28]
Hello,

First of all, I just wanna say that I really liked the library, and it’s functional approach.

I don’t know if is in the backlog, or the author have any intention, but it’d be great if had react-native support.

**Comment by christianalfoni** [comment]
@gsgualbano Thanks for the feedback! :) 

Support for react native is a bit of a scoping issue really, cause we have to explicitly support it in the core. Styled-components only supports React, so it is naturally within their scope to also support react native.

There is currently no natural hook to extend the core functionality into like: `@stitches/react-native`, cause we have to change how the core works. Not outputting css classnames and producing stylesheets, but rather use the low level stylesheet API of react native.

So basically this would require a refactor of the core and make it more extensible. 

Anyways, I am talking challenges. It is certainly possible, though will need to push it down the list a bit for now 😄

**Comment by matthova** [comment]
I'm thinking this would be doable. We would need an alternative api that turns style declarations into a style object. Then we feed the style object into a ReactNative Stylesheet declaration. We could even offer breakpoint handling if we offered an alternative window listener that tracked the window width and height.

I'd like to do some digging next weekend, but @christianalfoni if you know of a high-level outline of how we could get a style object without calling stylesheets, would love your input

**Comment by ecreeth** [comment]
I would ❤️ **stitches** in react native!

**Comment by appzillamonster** [comment]
I think this issue should stay open in order to track it.

**Comment by brodybits** [comment]
Would DOM in core (PR #450) be redundant with React Native support?

**Comment by peduarte** [comment]
@brodybits not sure how this affect React Native, but its on our roadmap to eventually have a RN-specific version.
